openmdao/core/group.py

```diff
@@ -1273,6 +1273,7 @@ def _solve_nonlinear(self):
         # TODO: It is pointless to run this ahead of non-iterative solvers.
         for sub in self.system_iter(recurse=True):
             if hasattr(sub, 'guess_nonlinear'):
+                self._transfer('nonlinear', 'fwd')
```

Comment:
Ah, right. The way I'd suggest implementing this would be to call `_guess_nonlinear`, instead of iterating at all. Then, `_guess_nonlinear` would have to be a method on group and component. Group's `_guess_nonlinear` would just call its children's `_guess_nonlinear`, and component's `_guess_nonlinear` would call `guess_nonlinear` in an unscaled context